contracts/Pool.sol

```diff
@@ -441,6 +445,42 @@ contract Pool is PoolFDT {
         _updateFundsTokenBalance();
     }
 
+    /**
+        @dev Increase the custody allowance for a given `custodian` corresponds to `msg.sender`.
+        @param custodian Address which will act as custodian of given `amount` for a tokenHolder.
+        @param amount    Number of FDTs cusodied by the custodian.
+     */
+    function increaseCustodyAllowance(address custodian, uint256 amount) external {
+        uint256 oldAllowance      = custodyAllowance[msg.sender][custodian];
+        uint256 newAllowance      = oldAllowance.add(amount);
+        uint256 newTotalAllowance = totalCustodyAllowance[msg.sender].add(amount);
+
+        PoolLib.increaseCustodyAllowanceChecks(custodian, amount, newTotalAllowance, balanceOf(msg.sender));
+        
+        custodyAllowance[msg.sender][custodian] = newAllowance;
+        totalCustodyAllowance[msg.sender]       = newTotalAllowance;
+        emit CustodyAllowanceChanged(msg.sender, custodian, oldAllowance, newAllowance);
+    }
+
+    /**
+        @dev Transfer custodied FDTs to `to` address.
+        @notice from and to should always be equal in this implementation.
```

Comment:
Sorry for the last pointer but we can update the dev comments here to clear it more for the external auditors.